pkg/volume/csi/csi_mounter.go

```diff
@@ -284,7 +284,7 @@ func (c *csiMountMgr) SetUpAt(dir string, mounterArgs volume.MounterArgs) error
 	defer func() {
 		// Only if there was an error and volume operation was considered
 		// finished, we should remove the directory.
-		if err != nil && volumetypes.IsOperationFinishedError(err) {
+		if retErr != nil && volumetypes.IsOperationFinishedError(retErr) {
```

Comment:
where is `retErr` getting set btw?

Reply:
@gnufied when the function is returned, the error will be set with the return value.

Reply:
```
(base) ➜  ~ go run main.go
a

(base) ➜  ~
(base) ➜  ~ cat main.go
package main

import "errors"
import "fmt"

func main() {
    f()
}

func f() (err error) {
    defer func(){ fmt.Println(err) }()
    return errors.New("a")
}
```

Reply:
Thank you. Though I am personally wary of automatic error variable assignment this way. Evidently, I had difficulty wrapping my head around it, so will anyone else who is reading this code.